tools/bwa.py

```diff
@@ -168,20 +170,44 @@ def align_mem_one_rg(self, inBam, refDb, outBam, rgid=None, options=None,
 
         assert len(readgroup_line) > 0
 
-        # rather than reheader the alignment bam file later so it has the readgroup information
-        # from the original bam file, we'll pass the RG line to bwa to write out
-        self.mem(one_rg_inBam, refDb, outBam, options=options+['-R',
-                 readgroup_line.rstrip("\r\n")],
-                 min_score_to_filter=min_score_to_filter, threads=threads, invert_filter=invert_filter)
-
-        # if there was more than one RG in the input, we had to create a temporary file with the one RG specified
-        # and we can safely delete it this file
-        # if there was only one RG in the input, we used it directly and should not delete it
-        if removeInput:
-            os.unlink(one_rg_inBam)
+        with util.file.tempfname('.aligned.bam') as tmp_bam_aligned:
+            # rather than reheader the alignment bam file later so it has the readgroup information
+            # from the original bam file, we'll pass the RG line to bwa to write out
+            self.mem(one_rg_inBam, refDb, tmp_bam_aligned, options=options+['-R',
+                     readgroup_line.rstrip("\r\n")],
```

Comment:
Replace tabs with with `\t` or else samtools will warn about duplicate IDs even though they are in the `@PG` header line later.